### PR TITLE
LL-2279 ~ LL-2335

### PR DESCRIPTION
**Pull request by LFBarreto**
(TRX): added base voting flow not editable yet but functional.

### Type

Feature

### Context

LL-2279 LL-2335

### Parts of the app affected / Test plan

TRX accounts that can vote

### CODE REVIEW

File: src/renderer/modals/VoteTron/Body.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+type StateProps = {|
+  t: TFunction,
+  device: ?Device,
+  accounts: Account[],
+  device: ?Device,
+  closeModal: string => void,
+  openModal: string => void,
+|};
+
+type Props = {|
```
Comment (LukeSugiura) [inline src/renderer/modals/VoteTron/Body.js]: i'm not a flow expert but can you just write like `type Props = OwnProps & StateProps`?

File: src/renderer/modals/VoteTron/Body.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  t,
+  stepId,
+  device,
+  closeModal,
+  openModal,
+  onChangeStepId,
+  params,
+  name,
+}: Props) => {
+  const [steps] = useState(createSteps);
```
Comment (LukeSugiura) [inline src/renderer/modals/VoteTron/Body.js]: useMemo?

File: src/renderer/modals/VoteTron/Body.js (hunk 1 above)
Comment (LFBarreto) [reply src/renderer/modals/VoteTron/Body.js]: probably i copy pasted this part from other modals :sweat_smile:

File: src/renderer/modals/VoteTron/Body.js (hunk 2 above)
Comment (LFBarreto) [reply src/renderer/modals/VoteTron/Body.js]: also copy pasted this without thinking but you're right on this :thinking:

File: src/renderer/modals/VoteTron/Body.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+  return (
+    <Stepper {...stepperProps}>
+      <SyncSkipUnderPriority priority={100} />
+      <Track onUnmount event="CloseModalVote" />
+    </Stepper>
+  );
+};
+
+const C: React$ComponentType<OwnProps> = compose(
```
Comment (LukeSugiura) [inline src/renderer/modals/VoteTron/Body.js]: Is there any reasons that this is not directly exported?

File: src/renderer/modals/VoteTron/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import type { StepId } from "./types";
+type State = {
+  stepId: StepId,
+};
+
+const INITIAL_STATE = {
+  stepId: "castVotes",
+};
+
+class VoteModal extends PureComponent<{ name: string }, State> {
```
Comment (LukeSugiura) [inline src/renderer/modals/VoteTron/index.js]: Why is this a class component?

File: src/renderer/modals/VoteTron/Body.js (hunk 3 above)
Comment (LFBarreto) [reply src/renderer/modals/VoteTron/Body.js]: I tried but i think its the typing on those compose components that are a bit tricky :thinking:

File: src/renderer/modals/VoteTron/index.js (hunk 4 above)
Comment (LFBarreto) [reply src/renderer/modals/VoteTron/index.js]: same copy pasted the same way of doing stepper modals we do elsewhere. :sweat_smile:

File: src/renderer/modals/VoteTron/fields/VotesField.js [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+      </IconContainer>
+      <VoteInput value={value} disabled={disabled} onChange={e => onChange(e.target.value)} />
+    </InputBox>
+  </Row>
+);
+
+const SR_THRESHOLD = 27;
+
+type Props = {
+  t: TFunction,
```
Comment (LukeSugiura) [inline src/renderer/modals/VoteTron/fields/VotesField.js]: I rather use useTranslation hook inside this component instead of passing as a prop.

Comment (LFBarreto) [reply src/renderer/modals/VoteTron/fields/VotesField.js]: I don't even use it i'll remove this :+1:

**Pull request merged**